File: ueueo-framework/ueueo-exception-handing/src/main/java/Abp/Http/Client/AbpRemoteCallException.cs
using System;
using System.Runtime.Serialization;
using Volo.Abp.ExceptionHandling;

namespace Volo.Abp.Http.Client;

[Serializable]
public class AbpRemoteCallException : AbpException, IHasErrorCode, IHasErrorDetails, IHasHttpStatusCode
{
    public int HttpStatusCode;// { get; set; }

    public String Code => Error?.Code;

    public String Details => Error?.Details;

    public RemoteServiceErrorInfo Error;// { get; set; }

    public AbpRemoteCallException()
    {

    }

    public AbpRemoteCallException(String message, Exception innerException = null)
        : base(message, innerException)
    {

    }

    public AbpRemoteCallException(SerializationInfo serializationInfo, StreamingContext context)
        : base(serializationInfo, context)
    {

    }

    public AbpRemoteCallException(RemoteServiceErrorInfo error, Exception innerException = null)
        : base(error.Message, innerException)
    {
        Error = error;

        if (error.Data != null)
        {
            for (var dataKey in error.Data.Keys)
            {
                Data[dataKey] = error.Data[dataKey];
            }
        }
    }
}

File: ueueo-framework/ueueo-exception-handing/src/main/java/com/ueueo/http/client/AbpRemoteCallException.java
package com.ueueo.http.client;

import com.ueueo.http.RemoteServiceErrorInfo;
import com.ueueo.AbpException;
import com.ueueo.exceptionhandling.IHasErrorCode;
import com.ueueo.exceptionhandling.IHasErrorDetails;
import com.ueueo.exceptionhandling.IHasHttpStatusCode;
import lombok.Data;

@Data
public class AbpRemoteCallException extends AbpException implements IHasErrorCode, IHasErrorDetails, IHasHttpStatusCode {

    private int httpStatusCode;

    private RemoteServiceErrorInfo error;

    @Override
    public String getCode() {
        return error != null ? error.getCode() : null;
    }

    @Override
    public String getDetails() {
        return error != null ? error.getDetails() : null;
    }

    public AbpRemoteCallException() {

    }

    public AbpRemoteCallException(String message, Exception innerException) {
        super(message, innerException);
    }

    public AbpRemoteCallException(RemoteServiceErrorInfo error, Exception innerException) {
        super(error.getMessage(), innerException);
        this.error = error;
        if (error.getData() != null) {
            this.data.putAll(error.getData());
        }
    }
}
